### ISSUE TITLE
0.12 broke API a little

### ISSUE THREAD

**Issue by yarikoptic** [joblib/joblib#728]
In 9d521102a9fbfd5fb3e7b90c7df9f5e0047ee8e6
BatchedCalls got 3rd required positional argument `pickle_cache`.  That apparently broke some dependent code in dask.distributed/1.21.8 . See http://bugs.debian.org/904944 for more detail
Do you see an easy way to retain compatibility?  if not, at least an API-breakage warning then should be added to CHANGES.md

Cheers!

**Comment by GaelVaroquaux** [comment]
Tell us more. What breaks what?

**Comment by yarikoptic** [comment]
Hi Gael, did you receive an empty email? I pressed enter too early, issue body was edited and visible above. It is a new required positional argument in BatchedCalls

**Comment by ogrisel** [comment]
Actually the bug is on dask distributed side: the `LooseVersion(joblib.__version__) <= "0.11.0"` test should be `LooseVersion(joblib.__version__) <= LooseVersion("0.11.0")`.

I fixed in joblib master (that now includes the dask connector code to simplify maintenance) but I can also make `pickle_cache` a kwarg to preserve backward compat. Will fix this for 0.12.1.

**Comment by yarikoptic** [comment]
Hi @ogrisel , thanks for the fix.

Just for my own clarity -- why do you (dask) needs LooseVersion on the right side? AFAIK and looking at the code in version.py:
```python
    def __cmp__ (self, other):
        if isinstance(other, StringType):
            other = LooseVersion(other)

        return cmp(self.version, other.version)
```
the right side is casted automatically to LooseVersion anyways
